### PR TITLE
Feat/gate-with-passport

### PR DESCRIPTION
**Pull request by delightfulabyss**
Adds NFT ownership check to sign-in process, which is then referenced before upvoting and link submission.

Test ERC721 contract is here: 

https://rinkeby.etherscan.io/address/0xadc637aa19edf5e9ed8088785d5c367248962a5a#readContract

### CODE REVIEW

File: packages/app/components/Post/Post.tsx [hunk 1]
```diff
@@ -132,12 +132,17 @@ const Post = ({
   const upvoteHandler = useCallback(
     (_id: string) => {
       if (isAuthenticated) {
-        return mutate({ postId: _id });
+        if(isPassportOwner){
+          return mutate({ postId: _id });
```
Comment (jhsu) [inline packages/app/components/Post/Post.tsx:142]: This check should probably be at the api upvote hander

File: packages/app/pages/api/me.ts [hunk 2]
```diff
@@ -1,12 +1,29 @@
 import { withIronSessionApiRoute } from 'iron-session/next';
 import { NextApiRequest, NextApiResponse } from 'next';
 import { ironOptions } from '@/constants';
+import  { Contract, ethers } from "ethers";
+
+const provider = new ethers.providers.AlchemyProvider("rinkeby", process.env.NEXT_PUBLIC_ALCHEMY_KEY);
```
Comment (jhsu) [inline packages/app/pages/api/me.ts:6]: i'm guessing this is set to rinkeby just for testing?

Comment (delightfulabyss) [reply packages/app/pages/api/me.ts:6]: Yup, we'll connect it to the real passport contracts eventually.

Comment (delightfulabyss) [reply packages/app/pages/api/me.ts:6]: Yup, just for testing.

File: packages/app/components/Post/Post.tsx (hunk 1 above)
Comment (delightfulabyss) [reply packages/app/components/Post/Post.tsx:142]: I was thinking that by storing the ownership status in global state after it's been derived either by the verify or passport routes, we minimize the number of external requests we need to make for checking token ownership status.  What do you think?

Comment (jhsu) [reply packages/app/components/Post/Post.tsx:142]: that's a good idea for the front-end side, but we still should authorize server-side

**Pull request merged**